File: packages/api-client/src/endpoints/health.ts
import type { ApiClient } from '../client/fetcher';

export interface HealthStatus {
  status: string;
}

export function createHealthEndpoints(client: ApiClient) {
  return {
    /** Unversioned infra probe — matches the actual FastAPI route (apps/api/app/api/health.py). */
    check: (): Promise<HealthStatus> => client.get<HealthStatus>('/health'),
  };
}

File: apps/api/app/api/health.py
from fastapi import APIRouter

from app.core.responses import SuccessResponse

router = APIRouter(tags=["health"])


@router.get("/health", response_model=SuccessResponse[dict[str, str]])
async def health_check() -> SuccessResponse[dict[str, str]]:
    """Unversioned infra health check, used by Docker/orchestrator probes."""
    return SuccessResponse(message="API is running.", data={"status": "ok"})
